### ISSUE TITLE
Verify ruby-lsp is working during run

### ISSUE THREAD

**Issue by noahgibbs** [Shopify/yjit-bench#239]
I'm seeing results from @eightbitraptor that look a *lot* like the ruby-lsp benchmark is failing, rapidly, and just not noticing, so it returns really fast, really unstable results.

So I figure I'll add some verification, to ensure that it's generating the right results, for some definition of that phrase. Probably "save the result from the last iteration, do a few basic checks on the value" sort of thing.

**Comment by noahgibbs** [comment]
I've updated the parser gem in the PR.

**Comment by vinistock** [comment]
The reason I included it is because running RuboCop related features, such as diagnostics, is by far the slowest operation the Ruby LSP performs - and YJIT helps a lot there.

But I did raise the point back when I added the benchmark if it would be better to split it into separate things and we ended up going with everything in the same benchmark.

I'm okay with splitting if people think it makes more sense. We can keep the `ruby-lsp` one for our regular operations and add a `ruby-lsp-rubocop` exclusively for RuboCop related stuff.

Once we integrate YARP into the Ruby LSP, the performance will change significantly. We can take that opportunity to both upgrade the Ruby LSP and then I'll split the RuboCop part into another benchmark. Does that sound good?
